src/maestro/workflow.py

```diff
@@ -468,6 +490,75 @@ def _initialize_opik(self) -> None:
         if self._opik is None:
             self._opik = Opik()
 
+    def _start_workflow_timing(self) -> None:
+        """Start timing the workflow execution."""
+        self.workflow_start_time = time.time()
+        self._timing_started = True
+
+    def _end_workflow_timing(self) -> None:
+        """End timing the workflow execution."""
+        if self._timing_started and self.workflow_end_time is None:
+            self.workflow_end_time = time.time()
+            self._timing_started = False
+
+    def force_end_timing(self) -> None:
+        """Force end timing if it's still running."""
+        if self._timing_started:
+            self._end_workflow_timing()
+
+    def _get_workflow_execution_time(self) -> float:
+        """Get the total workflow execution time in seconds."""
+        if self.workflow_start_time and self.workflow_end_time:
+            return self.workflow_end_time - self.workflow_start_time
+        return 0.0
+
+    def _track_agent_execution_time(
+        self, agent_name: str, execution_time: float
+    ) -> None:
+        """Track execution time for a specific agent."""
+        self.agent_execution_times[agent_name] = execution_time
+
+    def get_execution_metrics(self) -> Dict[str, Any]:
+        """Get execution time metrics for the workflow and all agents."""
+        if self._timing_started:
+            self.force_end_timing()
+
+        return {
+            "workflow_execution_time_seconds": self._get_workflow_execution_time(),
+            "agent_execution_times": self.agent_execution_times.copy(),
+            "total_agent_time_seconds": sum(self.agent_execution_times.values()),
+            "workflow_start_time": self.workflow_start_time,
+            "workflow_end_time": self.workflow_end_time,
+            "timing_status": "completed" if self.workflow_end_time else "running",
+        }
+
+    def get_token_usage_summary(self) -> Dict[str, Any]:
+        """Get token usage summary for all agents."""
+        total_token_usage = {
+            "total_prompt_tokens": 0,
+            "total_response_tokens": 0,
+            "total_tokens": 0,
+            "agent_token_usage": {},
+        }
+
+        for agent_name, agent in self.agents.items():
+            if hasattr(agent, "get_token_usage"):
+                token_usage = agent.get_token_usage()
+                total_token_usage["agent_token_usage"][agent_name] = token_usage
+
+                if "prompt_tokens" in token_usage:
```

Comment:
This seems to be duplicated also. Move to a common method and use. Avoid repeating yourself :)